### PR TITLE
Auto-download preview on complete.

### PR DESCRIPTION
**Pull request by jmcarp**
h/t @jehlers 

kusovzasby

PS: I'm kind of torn between the simplicity of rendering everything in the templates vs the granularity of procedural updates in javascript. Unfortunately, backbone doesn't include any kind of dom-diffing, virtual or otherwise, out of the box, so re-rendering the template on every change isn't blazingly fast--but managing all dom updates procedurally with callbacks and jquery is also a pain to think about and test. Thoughts @xtine @cmc333333 ?

### CODE REVIEW

**Comment by xtine** [comment]
Pro: re-rendering template makes for one less thing to try to make work without JS and accessibility reasons.

Con: slower, puts logic in template.

I vote for keeping as is for pro reasons.

File: regulations/static/regulations/js/source/views/comment/comment-review-view.js [hunk 1]
```diff
@@ -58,11 +55,13 @@ var CommentReviewView = Backbone.View.extend({
...
     });
     $xhr.then(this.previewSuccess.bind(this));
+    this.previewLoading = true;
+    this.render();
   },
 
   previewSuccess: function(resp) {
-    this.mode = 'submit';
-    this.previewUrl = resp.url;
+    window.location = resp.url;
```
Comment (cmc333333) [inline regulations/static/regulations/js/source/views/comment/comment-review-view.js]: Would it be possible to create a new tab instead? There's certainly an argument that the user expects to be "taken" to the file (despite Chrome's helpful stay-on-page-anyway approach), but it feels a little shady to do a `window.location`

File: regulations/static/regulations/js/source/views/comment/comment-review-view.js [hunk 2]
```diff
@@ -22,9 +22,7 @@ var CommentReviewView = Backbone.View.extend({
     this.docId = this.$el.data('doc-id');
     this.template = _.template($('#comment-template').html());
 
-    this.mode = 'preview';
-    this.previewUrl = null;
-
+    this.previewLoading = false;
```
Comment (cmc333333) [inline regulations/static/regulations/js/source/views/comment/comment-review-view.js]: Curious: can you imagine any race-cases between the the user's actions and this state? Would it make sense to pass this as a flag to `render` instead?

File: regulations/static/regulations/js/source/views/comment/comment-review-view.js (hunk 1 above)
Comment (jmcarp) [inline regulations/static/regulations/js/source/views/comment/comment-review-view.js]: That's a good point re. shadiness, but I'm not sure what else we can do here. I tried opening the link in a new tab instead:

```javascript
window.open(resp.url, '_blank');
```

But that triggered a pop-up warning in chrome that users might not notice and accept.

Also, do any modern browsers *not* open a download dialog on a response with a content-disposition of attachment?

File: regulations/static/regulations/js/source/views/comment/comment-review-view.js (hunk 2 above)
Comment (jmcarp) [inline regulations/static/regulations/js/source/views/comment/comment-review-view.js]: Don't think so--requesting a preview disables the preview button, so users shouldn't get racing preview requests unless they start messing with the console. Passing a loading flag to render could also lead to bugs--for example, if we allow users to delete comments from this view, deleting a comment would trigger a re-render, and if a preview was loading, we'd need to make sure to pass the flag to render. Unless we store loading state somewhere (javascript state, dom, looking at pending ajax requests), we're not going to know whether to render the preview button in loading or not-loading state at the point, unless I'm misunderstanding your suggestion.

File: regulations/static/regulations/js/source/views/comment/comment-review-view.js (hunk 1 above)
Comment (cmc333333) [inline regulations/static/regulations/js/source/views/comment/comment-review-view.js]: You've got it: http://greenbytes.de/tech/tc2231/#attonly

File: regulations/static/regulations/js/source/views/comment/comment-review-view.js (hunk 2 above)
Comment (cmc333333) [inline regulations/static/regulations/js/source/views/comment/comment-review-view.js]: Gotcha, agreed

**Comment by cmc333333** [comment]
:+1: x2

**Pull request merged**